lang/syn/src/parser/accounts/constraints.rs

```diff
@@ -494,6 +573,35 @@ impl<'ty> ConstraintGroupBuilder<'ty> {
         Ok(())
     }
 
+    // TODO are these the right errors to have?
+    fn add_mint_authority(&mut self, c: Context<ConstraintMintAuthority>) -> ParseResult<()> {
+        if self.mint_authority.is_some() {
+            return Err(ParseError::new(c.span(), "mint authority already provided"));
+        }
+        if self.mint_decimals.is_none() {
+            return Err(ParseError::new(
+                c.span(),
+                "mint decimals must be provided before authority",
```

Comment:
Debatable whether this should be enforced. :shrug:

Reply:
It made more sense with the old `token = <mint>, authority = <auth>` syntax, since having the `token` first made things more clear. But with the new syntax, it's not really necessary.

Reply:
I agree, there's no reason to enforce this.